python/ray/autoscaler/spark/cluster_init.py

```diff
@@ -0,0 +1,149 @@
+import copy
```

Comment:
This should be under ray/util/spark.

We should only have node provider related code under autoscaler folder